lib/Sema/CSSimplify.cpp

```diff
@@ -10185,6 +10229,41 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
     return recordFix(fix, impact) ? SolutionKind::Error : SolutionKind::Solved;
   }
 
+  case FixKind::TreatArrayLiteralAsDictionary: {
+    ArrayExpr *AE = getAsExpr<ArrayExpr>(fix->getAnchor());
+    assert(AE);
+
+    // If the array was empty, there's nothing to do.
+    if (AE->getNumElements() == 0)
+      return recordFix(fix) ? SolutionKind::Error : SolutionKind::Solved;
+
+    // For arrays with a single element, match it to the dictionary's key type.
+    auto elemTy = isArrayType(type1);
+    auto dictTy = type2->lookThroughAllOptionalTypes();
+    auto valueTy = createTypeVariable(getConstraintLocator(locator),
+                                      TVO_CanBindToLValue |
+                                        TVO_CanBindToNoEscape |
+                                        TVO_CanBindToHole);
```

Comment:
@xedin what are the right options here? I just copied these from elsewhere.

Reply:
That looks correct to me.